libraries/AP_OpticalFlow/OpticalFlow.h

```diff
@@ -36,6 +36,11 @@ class OpticalFlow
     OpticalFlow(const OpticalFlow &other) = delete;
     OpticalFlow &operator=(const OpticalFlow&) = delete;
 
+    // get singleton instance
+    static OpticalFlow *get_singleton() {
+        return _singleton;
```

Comment:
Just a note that this is not really a singleton pattern... For a real singleton you'd have:

OpticalFlow.cpp:
```
OpticalFlow OpticalFlow::_singleton;
```

Then the method would return `&singleton`.

The difference is that with this approach we are implicitly forcing the object to be constructed when someone tries to get it. In your current approach, a NULL pointer can still be returned if the object wasn't constructed by other means when this function is called.